seismostats/analysis/estimate_mc.py

```diff
@@ -318,3 +318,86 @@ def mc_max_curvature(
     )
     mc = bins[count.argmax()] + correction_factor
     return mc
+
+
+def mc_by_bvalue_stability(
+        sample: np.ndarray, delta_m: float, stability_factor: float = 0.1):
+    """
+    Estimates Mc using a test of stability.
+    The stability of the b-value is tested by default on half a magnitude unit
+    (in line with the 5x0.1 in the orginial paper). Users can change the range
+    for the stability test by changing the stability_factor (which gives the
+    size of the magnitude bin to use).
+
+    Source:
+        Cao, A., & Gao, S. S. (2002). Temporal variation of seismic b-values
+            beneath northeastern Japan island arc. Geophysical Research
+            Letters, 29(9), 1–3. https://doi.org/10.1029/2001gl013775
+
+    Args:
+        sample : np.array Vector of magnitudes
+        delta_m : float. Discretization of the magnitudes.
+        stability_factor : float. Magnitude bin to consider for the
```

Comment:
I would prefer to use a stability_range- and the binning would be used as is given by the delta_m.

The stability_range would be default at 0.5, but could be any value, (e.g., 0.2), and stability_range/delta_m points will be evaluated for stability. E.g.,  for stability_range = 0.6 and delta_m = 0.1, six points will be evaluated, but if delta_m= 0.2, only 3 points.

This would make the parameter stability_range more meaningful. What do you think?

Reply:
Would also be very easy to implement I think :)